### ISSUE TITLE
International characters do not display in Slack alerts

### ISSUE THREAD

**Issue by zbodi74** [metabase/metabase#38753]
### Describe the bug

Unicode characters not showing up in Slack alerts, known to affect at least these versions: v1.47.9, v1.48.3 and v1.48.5.
In email notifications the text displays as expected.

### To Reproduce

Create a native question that returns text with international characters, and set up a Slack alert for either the card or the  dashboard.

```SELECT 'هذا نص عربي عشوائي، يستخدم لأغراض توضيحية فقط. لا يحتوي على أي معنى محدد ومصمم ليكون غير مسيء.' AS Text, 'Arabic' AS Language
SELECT '这是一个随机的中文文本，仅用于说明目的。它没有具体的意义，旨在不冒犯。', 'Chinese'
UNION ALL
SELECT 'זהו טקסט עברי אקראי, המשמש למטרות המחשה בלבד. אינו מכיל משמעות מסוימת ומעוצב להיות לא מעליב.', 'Hebrew'
UNION ALL
SELECT 'árvíztűrő tükörfúrógép', 'Hungarian'
UNION ALL
SELECT '이것은 임의의 한국어 텍스트로, 설명 목적으로만 사용됩니다. 구체적인 의미를 포함하지 않으며 불쾌감을 주지 않도록 설계되었습니다.', 'Korean'
```

Results in the browser:
<img width="676" alt="image" src="https://github.com/metabase/metabase/assets/11080638/2800a35e-fb40-43e0-8dd9-cbc6adfd453b">

Results in Email:
<img width="590" alt="image" src="https://github.com/metabase/metabase/assets/11080638/52890a6c-79ae-4db6-8401-c34a004442ee">

Results in Slack:
<img width="390" alt="image" src="https://github.com/metabase/metabase/assets/11080638/187d8699-a78c-42ba-bdf8-62dff1c64258">

### Expected behavior

All characters should be rendered correctly

### Logs

n/a

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.22+7",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.22",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.22+7",
    "os.name": "Linux",
    "os.version": "5.10.209-198.812.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "snowflake",
      "bigquery-cloud-sdk",
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "14.7"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.6.0"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-02-06",
      "tag": "v1.48.5",
      "hash": "dab12cf"
    },
    "settings": {
      "report-timezone": "America/Los_Angeles"
    }
  }
}
```

### Severity

P1 - no workaround, reported by a customer

**Comment by paoliniluis** [comment]
Have you pulled the fonts packages in your environment?

**Comment by zbodi74** [comment]
@paoliniluis - yes, I reproduced this on a cloud instance as well as with Docker using the released image `metabase/metabase-enterprise:v1.48.5`.

**Comment by camsaul** [comment]
Regression of #2027

**Comment by darksciencebase** [comment]
confirmed it's still happening on master as of today (i used stats so it's not my environment)

**Comment by qnkhuat** [comment]
I think this is a CSSBox's bug see: https://github.com/radkovo/CSSBox/issues/85

TLDR:
- We use Lato as our primary font, and Lato does not support non-Latin languages.
- So when rendering a Korean text with lato, it's invalid and it should fall back to the next font family, which is defined [here](https://github.com/metabase/metabase/blob/44fb3171f47877c5edabc2b76de466ae52a4795e/src/metabase/pulse/render/style.clj#L76)
- In order to send charts to Slack, we render those charts from HTML to PNG using CSSBox (see [this](https://github.com/metabase/metabase/blob/44fb3171f47877c5edabc2b76de466ae52a4795e/src/metabase/pulse/render/png.clj#L76) code)
- CSSBox is not handling font fallback correctly

There are a couple of options:
1. We should try to fix the CSSBox bug
2. Change from Lato to [Noto](https://fonts.google.com/noto) - a font that supports a wide range of language

**Comment by yanghua-ola** [comment]
> I think this is a CSSBox's bug see: [radkovo/CSSBox#85](https://github.com/radkovo/CSSBox/issues/85)
> 
> TLDR:
> 
> * We use Lato as our primary font, and Lato does not support non-Latin languages.
> * So when rendering a Korean text with lato, it's invalid and it should fall back to the next font family, which is defined [here](https://github.com/metabase/metabase/blob/44fb3171f47877c5edabc2b76de466ae52a4795e/src/metabase/pulse/render/style.clj#L76)
> * In order to send charts to Slack, we render those charts from HTML to PNG using CSSBox (see [this](https://github.com/metabase/metabase/blob/44fb3171f47877c5edabc2b76de466ae52a4795e/src/metabase/pulse/render/png.clj#L76) code)
> * CSSBox is not handling font fallback correctly
> 
> There are a couple of options:
> 
> 1. We should try to fix the CSSBox bug
> 2. Change from Lato to [Noto](https://fonts.google.com/noto) - a font that supports a wide range of language

Any plans on the fix? Any workaround before it is fixed?

I tried to alias Noto Sans as Lato on system level; but it appeared metabase was using an embedded font file `frontend_client/app/fonts/Lato/lato-v16-latin-%s.ttf`. Thus it didn't work.

**Comment by adam-james-v** [comment]
@yanghua-ola , I have a fix that is up and ready for review. Hopefully we can get that rolling next week.
Unfortunately, I'm not sure of a helpful workaround for you at this time, especially if loading your own fonts in the system hasn't worked

**Comment by Tony-metabase** [comment]
There is still a missing piece here since it works for Tables but not for other charts. Lik a Bar chart:

![image](https://github.com/user-attachments/assets/e5fddcc1-a343-4c1e-9a43-30627e597dfc)

Subscription:

![image](https://github.com/user-attachments/assets/c30a4e5e-9e2c-46be-a0de-49fa95b644a1)

**Comment by Tony-metabase** [comment]
Closing this one out since the fix was not properly applied to the whole library. I opened a separate issue which should take care of the charts:

https://github.com/metabase/metabase/issues/47497
